staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -1221,9 +1221,15 @@ func (vs *VSphere) DisksAreAttached(nodeVolumes map[k8stypes.NodeName][]string)
 			}
 
 		}
+		klog.V(4).Infof("DisksAreAttached successfully executed. result: %+v", attached)
+		// There could be nodes in cluster which do not have any pods with vsphere volumes running on them
+		// such nodes won't be part of nodeVolumes map because attach-detach controller does not keep track
+		// such nodes. But such nodes may still have dangling volumes on them and hence we need to scan all the
+		// remaining nodes which weren't scanned by code previously.
+		vs.BuildMissingVolumeNodeMap(ctx)
```

Comment:
How expensive is this?  Is this method part of a periodic check, or invoked once per disk attach/detach?

Reply:
This check is performed periodically. The check should not be more expensive than what we were doing already. We were already scanning all the nodes (although grouped by vcenter) for disks and reconciling them for disks that should be attached to a node but somehow is detached. This mechanism exists for all cloudproviders.

What this change is doing is kind of other way around. As part of same loop (which already runs), it will periodically scan for disks that shouldn't be attached to a node but *somehow are* (say because of a crashed node) and it ensures that attach-detach-controller knows about such disks.